src/main/java/bio/terra/service/filedata/azure/tables/TableDirectoryDao.java

```diff
@@ -219,11 +224,11 @@ public FireStoreDirectoryEntry retrieveByPath(
       TableServiceClient tableServiceClient,
       UUID collectionId,
       String tableName,
-      List<String> fullPaths) {
+      Set<String> fullPaths) {
```

Comment:
We don't do this much, but another approach is to use a contravariant parameter type, e.g. you specify a more general subclass to allow callers to pass in a more specific type. In this case you'd declare this function to accept `Collection<String>`. That way a caller could provide `List` or `Set` and this code wouldn't need to be changed.

Reply:
That's a good idea - I've switched the items that didn't need to be switched to a set back to a list and used a collection here.